backend/src/main/java/fk/prof/backend/http/BackendHttpVerticle.java

```diff
@@ -290,20 +307,23 @@ private void handleGetAssociations(RoutingContext context) {
     }
   }
 
-  private Future<ProfHttpClient.ResponseWithStatusTuple> makeRequestPostAssociation(BackendDTO.LeaderDetail leaderDetail, Recorder.RecorderInfo payload)
-      throws IOException {
-    Buffer payloadAsBuffer = ProtoUtil.buildBufferFromProto(payload);
-    return httpClient.requestAsyncWithRetry(
-        HttpMethod.POST,
-        leaderDetail.getHost(), leaderDetail.getPort(), ApiPathConstants.LEADER_POST_ASSOCIATION,
-        payloadAsBuffer);
+  private Future<ProfHttpClient.ResponseWithStatusTuple> makeRequestToLeader(BackendDTO.LeaderDetail leaderDetail, HttpMethod method, String path, Buffer payloadAsBuffer, boolean withRetry) {
+    path = ApiPathConstants.LEADER + path;
+    if(withRetry){
+      return httpClient.requestAsyncWithRetry(method, leaderDetail.getHost(), leaderDetail.getPort(), path, payloadAsBuffer);
+    }else{
```

Comment:
Apparently the diff is not that significant for github to show it as a replacement instead it is showing it as formatting changes.